addons/account/models/account_payment.py

```diff
@@ -182,20 +182,26 @@ def _seek_for_lines(self):
         '''
         self.ensure_one()
 
-        liquidity_lines = self.env['account.move.line']
-        counterpart_lines = self.env['account.move.line']
-        writeoff_lines = self.env['account.move.line']
-
+        # liquidity_lines, counterpart_lines, writeoff_lines
+        lines = [self.env['account.move.line'] for _dummy in range(3)]
         valid_account_types = self._get_valid_payment_account_types()
         for line in self.move_id.line_ids:
             if line.account_id in self._get_valid_liquidity_accounts():
-                liquidity_lines += line
-            elif line.account_id.account_type in valid_account_types or line.account_id == self.company_id.transfer_account_id:
-                counterpart_lines += line
+                lines[0] += line  # liquidity_lines
+            elif line.account_id.account_type in valid_account_types or line.account_id == line.company_id.transfer_account_id:
+                lines[1] += line  # counterpart_lines
             else:
-                writeoff_lines += line
+                lines[2] += line  # writeoff_lines
+
+        # In some case, there is no liquidity or counterpart line (after changing an outstanding account on the journal for example)
+        # In that case, and if there is one writeoff line, we take this line and set it as liquidity/counterpart line
+        if len(lines[2]) == 1:
+            for i in (0, 1):
+                if not lines[i]:
+                    lines[i] = lines[2]
+                    lines[2] -= lines[2]
```

Comment:
this is a fw port of https://github.com/odoo/odoo/pull/149285 which is not needed in 16 (I didn't manage to find why exactly), and needed here only where we reset move to draft/unreconcile

> But this will fail if for some reason you have neither liquidity_lines nor counterpart_lines, if that is even possible

Yes I know, we've already limited the case by checking if `len(lines) == 2` but as you said, having neither liquidity or counterpart should not happen, or fail if it is the case.

Reply:
Mention in the PR description that it is a fw-port :wink: